roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/util/ConfigParserTest.java

```diff
@@ -117,13 +121,95 @@ public void configParserTest_generateModel() throws ConfigParserException {
     }
 
     @Test
-    public void configParserTest_connectorsHaveData() throws ConfigParserException {
+    public void configParserTest_structureConnectorsHaveData() throws ConfigParserException {
         Experiment exp = new Experiment("Experiment");
-        IRoundaboutStructure roundaboutStructure = configParser.generateStructure(roundAboutConfig, exp);
+        IRoundaboutStructure roundaboutStructure = configParser.generateRoundaboutStructure(roundAboutConfig, exp);
 
         for (IStreetConnector connector : roundaboutStructure.getStreetConnectors()) {
             assertNotEquals("next street sections not empty", 0, connector.getNextSections().size());
             assertNotEquals("previous street sections not empty", 0, connector.getPreviousSections().size());
         }
     }
+
+    @Test
+    public void configParserTest_structureParameterSet() throws ConfigParserException {
+        Experiment exp = new Experiment("Experiment");
+        IRoundaboutStructure roundaboutStructure = configParser.generateRoundaboutStructure(roundAboutConfig, exp);
+
+        assertNotNull("parameters map not empty", roundaboutStructure.getParameters());
+        assertEquals("5 parameters read", 5, roundaboutStructure.getParameters().size());
+        assertTrue("MAIN_ARRIVAL_RATE_FOR_ONEWAYSTREETS in map", roundaboutStructure.getParameters().keySet().contains("MAIN_ARRIVAL_RATE_FOR_ONEWAYSTREETS"));
+        assertEquals("value of MAIN_ARRIVAL_RATE_FOR_ONEWAYSTREETS correct", "1.0", roundaboutStructure.getParameters().get("MAIN_ARRIVAL_RATE_FOR_ONEWAYSTREETS"));
+    }
```

Comment:
i know its easy this way, but consider this:
if the first assert fails, you do not know the state of the other (maybe) important information/asserts. if you would write on test for every assert you will be able to find an error much faster

info according multiple asserts: http://www.owenpellegrin.com/blog/testing/how-do-you-solve-multiple-asserts/

Reply:
true, but they depend on each other, so when one fails, the next won't work ... if they do not depend on each other, separation would be a lot better, you're right